crypto/core_namemap.c

```diff
@@ -137,7 +138,8 @@ void ossl_namemap_doall_names(const OSSL_NAMEMAP *namemap, int number,
     cbdata.number = number;
     cbdata.fn = fn;
     cbdata.data = data;
-    CRYPTO_THREAD_read_lock(namemap->lock);
+    if (!CRYPTO_THREAD_read_lock(namemap->lock))
+        return;
```

Comment:
This should probably be changed to indicate success or failure.

Reply:
> This should probably be changed to indicate success or failure.

the "do_all" stuff doesn't support that sadly.

Reply:
As of d84f5515f it does.

Reply:
thanks for the update, @mattcaswell .  So should this function be changed to return an error?  If so, 0/1 or -1/1 ?

Reply:
In latest master this function can fail (for other reasons besides locking) and either returns 0 for failure or 1 for success.

Reply:
> In latest master this function can fail (for other reasons besides locking) and either returns 0 for failure or 1 for success.

turns out the code in this PR already does that, from when I recently rebased.